### PR TITLE
Updated to latest stable available packages

### PR DESCRIPTION
**Pull request by concatime**
* ~~Update LibUV to 1.18.0~~
* Update LuaJIT to 2.0.5 (latest commit from branch 2.0)
* Update msgPack-C to 2.1.5
* Update LuaRocks to 2.4.3
* Update Jemalloc to 5.0.1
* Update WinPTY to 0.4.3

Update#1: Reverting LibUV to `1.12.0`. See #7506.

### CODE REVIEW

File: third-party/CMakeLists.txt [hunk 1]
```diff
@@ -90,18 +90,18 @@ include(ExternalProject)
...
+set(LUAJIT_SHA256 cbae019b5e396164eb5f0d07777b55cc03931bb944f83c61a010c053c9f5fd5b)
 
 set(LUA_URL https://www.lua.org/ftp/lua-5.1.5.tar.gz)
 set(LUA_SHA256 2640fc56a795f29d28ef15e13c34a47e223960b0240e8cb0a82d9b0738695333)
 
 # NOTE: Version must match LUAROCKS_VERSION in third-party/cmake/BuildLuarocks.cmake
-set(LUAROCKS_URL https://github.com/luarocks/luarocks/archive/v2.4.2.tar.gz)
-set(LUAROCKS_SHA256 eef88c2429c715a7beb921e4b1ba571dddb7c74a250fbb0d3cc0d4be7a5865d9)
+set(LUAROCKS_URL https://github.com/luarocks/luarocks/archive/v2.4.3.tar.gz)
+set(LUAROCKS_SHA256 ea1881d6954f2a98c34f93674571c8f0cbdbc28dedb3fa3cb56b6a91886d1a99)
```
Comment (justinmk) [inline third-party/CMakeLists.txt]: BuildLuarocks.cmake must also be updated, see the comment above.

Comment (concatime) [reply third-party/CMakeLists.txt]: Ok, corrected.

File: third-party/cmake/BuildLuarocks.cmake [hunk 2]
```diff
@@ -42,7 +42,7 @@ endfunction()
 set(LUAROCKS_BINARY ${HOSTDEPS_BIN_DIR}/luarocks)
 
 # NOTE: Version must match version of LuaRocks in third-party/CMakeLists.txt
-set(LUAROCKS_VERSION 2.4)
+set(LUAROCKS_VERSION 2.4.3)
```
Comment (justinmk) [inline third-party/cmake/BuildLuarocks.cmake]: Hmm maybe it doesn't need an update if only the patch number changed. Better leave this alone, unless CI complains.

Ultimately if CI passes then it's fine. I will also restart the CI after clearing the caches, before merging this.

**Comment by concatime** [comment]
Summary:

* QuickBuild is failing on FreeBSD because of
`Error parsing TAP Stream: Missing TAP Plan. caused by: Error parsing TAP Stream: Missing TAP Plan. caused by: Missing TAP Plan.`

* TravisCI is hanging…
`No output has been received in the last 10m0s`

**Comment by jamessan** [comment]
The QB failure is a known flaky test.  Travis was having issues for an hour or so.  I've restarted it to see how things go now.

**Comment by jamessan** [comment]
ASAN failure is known and fixed by https://github.com/neovim/neovim/pull/7739.  The remaining Travis jobs have been started manually.

**Review by jamessan — APPROVED** [review]

File: third-party/cmake/BuildLuarocks.cmake (hunk 2 above)
Comment (justinmk) [reply third-party/cmake/BuildLuarocks.cmake]: @concatime would you mind removing this change? Better to leave things unchanged if they aren't needed.

**Comment by justinmk** [comment]
Merged. Thanks

**Comment by fwalch** [comment]
Merging this broke the Ubuntu Unstable PPA builds on arm64 (segfault when running `vimball-tags`; see e.g. https://launchpadlibrarian.net/351647411/buildlog_ubuntu-artful-arm64.neovim_0.2.0ubuntu1+git201712291800+3837+26~ubuntu17.10.1_BUILDING.txt.gz). Worked around for now by forcing deps to https://github.com/neovim/deps/commit/0e83b08646a991259d109b3ecbc06ee944f8ceb7 in https://code.launchpad.net/~neovim-ppa/+recipe/neovim-unstable.

**Comment by justinmk** [comment]
@fwalch could you maybe bisect to see which dep causes the issue? 

```
make[5]: Entering directory '/<<BUILDDIR>>/neovim-0.2.0ubuntu1+git201712291800+3837+26~ubuntu17.10.1/build'
Segmentation fault (core dumped)
runtime/CMakeFiles/vimball-tags.dir/build.make:57: recipe for target 'runtime/CMakeFiles/vimball-tags' failed
make[5]: *** [runtime/CMakeFiles/vimball-tags] Error 139
make[5]: Leaving directory '/<<BUILDDIR>>/neovim-0.2.0ubuntu1+git201712291800+3837+26~ubuntu17.10.1/build'
CMakeFiles/Makefile2:7467: recipe for target 'runtime/CMakeFiles/vimball-tags.dir/all' failed
```

**Comment by fwalch** [comment]
[Reverting jemalloc to 4.5.0](https://code.launchpad.net/~neovim-ppa/neovim-ppa/+git/deps/+ref/jemalloc-4.5.0) while keeping the other dependency updates worked. Probably this has already been fixed on the jemalloc dev branch (https://github.com/jemalloc/jemalloc/issues/979).

**Comment by concatime** [comment]
Solutions:
1. Use the latest commit from `jemalloc`
2. Revert to `4.5.0`

**Comment by justinmk** [comment]
@concatime "Use the latest commit from jemalloc" is ok until a release is posted.

**Comment by concatime** [comment]
Do you want from me to create a new pull request?

**Comment by justinmk** [comment]
Sure

**Comment by justinmk** [comment]
@fwalch HEAD is revert to jemalloc 4.5: https://github.com/neovim/neovim/commit/a18db72a36cbd691e08283e7fc5557c4943300d4

**Comment by fwalch** [comment]
@justinmk Thanks for the heads up, I changed the Launchpad recipe back to use the deps `master` branch.

**Pull request closed without merging**